docs/notes/05-linters-and-tests.md

```diff
@@ -0,0 +1,179 @@
+
+### Benefits of Linters
+
+- **Improved Code Quality**: Linters help catch common programming mistakes, maintain consistent coding style, and enforce best practices, leading to cleaner and more reliable code.
+
+- **Early Bug Detection**: By analyzing code statically, linters can identify potential issues before the code is executed, enabling early bug detection and reducing the likelihood of bugs reaching production.
+
+- **Readability and Maintainability**: Linters promote consistent coding style and naming conventions, making the code more readable and maintainable. This improves collaboration among team members and makes it easier to understand and modify the codebase.
+
+- **Codebase Standardization**: Linters enforce coding standards and guidelines across the entire codebase, ensuring a unified coding style and reducing the risk of introducing code inconsistencies or errors.
+
+### Linters Requirements
+
+We require the following linters to be run on the code and included as checks in continuous integration:
+
+- [black](https://black.readthedocs.io/en/stable/)
+- [flake8](https://flake8.pycqa.org/en/latest/): Run command `flake8 src/ccn-template/ --max-complexity 10`
+- [isort](https://pycqa.github.io/isort/)
+
+## Tests
+
+Tests are an essential part of the software development process. They verify the correctness and expected behavior of 
+the code, ensuring that it performs as intended. Tests help identify bugs, ensure code quality, and provide confidence 
+in the reliability of the software.
+
+### Benefits of Tests
+
+- **Bug Detection**: Tests help identify bugs early in the development cycle, allowing developers to fix them before they reach production. By executing various scenarios, edge cases, and boundary conditions, tests increase the likelihood of discovering potential issues.
+
+- **Regression Prevention**: When modifications are made to the codebase, tests act as a safety net by ensuring that existing functionality remains intact. Tests catch regressions, preventing the reintroduction of bugs when new features or changes are introduced.
+
+- **Code Documentation**: Test cases act as a form of documentation, illustrating how different components of the codebase should behave and providing real-world examples of their usage. This helps improve code comprehension and assists developers in understanding how to interact with the code.
+
+- **Codebase Maintainability**: Tests promote codebase maintainability by facilitating refactoring and code evolution. They provide confidence that changes do not introduce unintended side effects or break existing functionality, enabling developers to make improvements with ease.
+
+### Standard Tests and Use Cases
+
+Below, we provide a non-comprehensive list of test types along with their general use cases and some examples.
```

Comment:
Any helpful links we can add here? How to write tests well is a ... big topic.

Reply:
Maybe [this one](https://www.fuzzingbook.org/#A-Textbook-for-Paper,-Screen,-and-Keyboard)?

Reply:
added this, another website, and a couple of books with titles and content that seemed appropriate